src/controller/editingcontroller.js

```diff
@@ -84,28 +85,64 @@ export default class EditingController {
 		 * @private
 		 * @member {utils.EmitterMixin} #_listenter
 		 */
-		this._listenter = Object.create( EmitterMixin );
+		this._listener = Object.create( EmitterMixin );
 
 		// Convert changes in model to view.
-		this._listenter.listenTo( this.model, 'change', ( evt, type, changes ) => {
+		this._listener.listenTo( this.model, 'change', ( evt, type, changes ) => {
 			this.modelToView.convertChange( type, changes );
 		}, { priority: 'low' } );
 
 		// Convert model selection to view.
-		this._listenter.listenTo( this.model, 'changesDone', () => {
+		this._listener.listenTo( this.model, 'changesDone', () => {
 			this.modelToView.convertSelection( model.selection );
 			this.view.render();
 		}, { priority: 'low' } );
 
+		// Convert model markers changes.
+		this._listener.listenTo( this.model.markers, 'addMarker', ( evt, name, range ) => {
+			this.modelToView.convertMarker( 'addMarker', name, range );
+		} );
+
+		this._listener.listenTo( this.model.markers, 'removeMarker', ( evt, name, range ) => {
+			this.modelToView.convertMarker( 'removeMarker', name, range );
+		} );
+
 		// Convert view selection to model.
-		this._listenter.listenTo( this.view, 'selectionChange', convertSelectionChange( model, this.mapper ) );
+		this._listener.listenTo( this.view, 'selectionChange', convertSelectionChange( model, this.mapper ) );
 
 		// Attach default content converters.
 		this.modelToView.on( 'insert:$text', insertText(), { priority: 'lowest' } );
 		this.modelToView.on( 'remove', remove(), { priority: 'low' } );
 		this.modelToView.on( 'move', move(), { priority: 'low' } );
 		this.modelToView.on( 'rename', rename(), { priority: 'low' } );
 
+		// Attach markers converters.
+		this.modelToView.on( 'insert', ( evt, data ) => {
+			const pos = data.range.start;
+
+			for ( let [ range, name ] of this.model.markers._markerNames ) {
+				if ( range.containsPosition( pos ) || range.start.isEqual( pos ) ) {
+					this.modelToView.convertMarker( 'addMarker', name, data.range );
+				}
+			}
+		}, { priority: 'lowest' } );
+
+		this.modelToView.on( 'move', ( evt, data ) => {
+			const pos = data.sourcePosition._getTransformedByInsertion( data.targetPosition, data.item.offsetSize );
```

Comment:
Checking with the start is because when you have a (marker) range and then you "cut" something out of it, the transformed range always starts or ends on sourcePosition. Examples:

`ab[cd{e]f}g` -> move `{}` -> `ab[cd]^g` -> `sourcePosition` will be after `d`
`a{b[c}de]fg` -> move `{}` -> `a^[de]fg` -> `sourcePosition` will be after `a`

That's why it makes sense to check `marker.range.start` and `marker.range.end`. But I will take a look if that loop cannot be simplified.